File: DSA/Cpp/test/dsa/container/lists/queues/my_stack_test.cpp
#include "queue_test.h"
#include <dsa/container/lists/queues/my_stack.hpp>

namespace dsa::container::lists::queues {
namespace {

class MyStackTest : public QueueTest {
 protected:
    MyStack<int> stack;
};
TEST_F(MyStackTest, emptyStack) {
    EXPECT_TRUE(stack.empty());
}
TEST_F(MyStackTest, pushNotEmpty) {
    stack.push(1);
    EXPECT_FALSE(stack.empty());
}
TEST_F(MyStackTest, sizeEmpty) {
    EXPECT_EQ(0, stack.size());
}
TEST_F(MyStackTest, pushSize) {
    stack.push(1);
    stack.push(2);
    EXPECT_EQ(2, stack.size());
}
TEST_F(MyStackTest, pushTop) {
    stack.push(1);
    stack.push(2);
    EXPECT_EQ(2, stack.top());
}
TEST_F(MyStackTest, pushPeek) {
    stack.push(1);
    stack.push(2);
    EXPECT_EQ(2, stack.peek());
}
TEST_F(MyStackTest, pushPop) {
    stack.push(1);
    stack.push(2);
    EXPECT_EQ(2, stack.pop());
}

}
};

File: DSA/Java/test/dsa/container/lists/queues/MyStackTest.java
package dsa.container.lists.queues;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.*;

public class MyStackTest extends MyStackTestFX {

    private MyStack<Integer> stack;

    @Before
    public void setUp() {
        stack = new MyStack<>();
    }

    @Test
    public void emptyStack() {
        assertTrue(stack.empty());
    }

    @Test
    public void pushNotEmpty() {
        stack.push(1);
        assertFalse(stack.empty());
    }

    @Test
    public void pushPopEmpty() {
        stack.push(1);
        assertEquals(Integer.valueOf(1), stack.pop());
        assertTrue(stack.empty());
    }

    @Test
    public void pushTop() {
        stack.push(1);
        stack.push(2);
        assertEquals(Integer.valueOf(2), stack.top());
    }

    @Test
    public void pushPeek() {
        stack.push(1);
        stack.push(2);
        assertEquals(Integer.valueOf(2), stack.peek());
    }
}